assembly/src/parsers/ast/context.rs

```diff
@@ -158,8 +158,17 @@ impl ParserContext {
             let proc_name_hash = self.get_proc_name_hash(full_proc_name);
             Ok(Node::Instruction(Instruction::ExecImported(proc_name_hash)))
         } else {
-            let index = self.procedures.get(&label).unwrap().index;
-            Ok(Node::Instruction(Instruction::ExecLocal(index)))
+            let procedure = self.local_procs.get(&label);
+            let index = match procedure {
+                Some(p) => p.0,
+                None => {
+                    return Err(AssemblyError::invalid_proc_label(
+                        tokens.read().unwrap(),
+                        &label,
```

Comment:
Shouldn't this be `undefined_proc` error? Also, I'd prefer to use `map_err` instead of `match` here.